client/post-editor/editor-html-toolbar/index.jsx

```diff
@@ -161,7 +161,24 @@ export class EditorHtmlToolbar extends Component {
 			previousSelectionEnd + insertedContentLength;
 	}
 
-	updateEditorContent( newContent ) {
+	updateEditorContent( before, newContent, after ) {
+		// execCommand( 'insertText' ), needed to preserve the undo stack, does not exist in IE11.
+		// Using the previous version of replacing the entire content value instead.
+		if ( !! window.MSInputMethodContext && !! document.documentMode ) {
+			this.updateEditorContentIE11( before + newContent + after );
+		} else {
+			this.insertEditorContent( before, newContent, after );
+		}
+	}
```

Comment:
since we don't expect the browser to change while someone is editing text I would recommend that we detect this on boot and use the resultant function in direct calls without the runtime `if`

```js
const isIEDetected = () = every( [
	window,
	document,
	window.MSInputMethodContext,
	document.documentMode,
] );

updateEditorContent = isIEDetected
	? ( before, newContent, after ) => this.updateEditorContentIE11( before + newContent + after )
	: ( before, newContent, after ) => this.insertEditorContent( before, newContent, after );
```